flask_restful/__init__.py

```diff
@@ -417,20 +430,28 @@ def url_for(self, resource, **values):
     def make_response(self, data, *args, **kwargs):
         """Looks up the representation transformer for the requested media
         type, invoking the transformer to create a response object. This
-        defaults to (application/json) if no transformer is found for the
-        requested mediatype.
+        defaults to default_mediatype if no transformer is found for the
+        requested mediatype. If default_mediatype is None, a 406 Not 
+        Acceptable response will be sent as per RFC 2616 section 14.1
 
         :param data: Python object containing response data to be transformed
         """
-        for mediatype in self.mediatypes() + [self.default_mediatype]:
-            if mediatype in self.representations:
-                resp = self.representations[mediatype](data, *args, **kwargs)
-                resp.headers['Content-Type'] = mediatype
-                return resp
+        default_mediatype = kwargs.pop('fallback_mediatype', None) or self.default_mediatype
+        mediatype = request.accept_mimetypes.best_match(
+            self.representations, 
+            default=default_mediatype,
+        )
+        if mediatype is None:
+            raise NotAcceptable()
+        if mediatype in self.representations:
```

Comment:
I think you are correct, returning `None` from `make_response` would cause a 500 error (I think the last test case you wrote shows that since raising a non `HTTPException` would cause a 500 error). I think I agree with you that 500 is more appropriate in that case.

I don't think adding the valid representations to the response would be particularly useful at this point since [`handle_errors`](https://github.com/flask-restful/flask-restful/blob/bf983cf706c9f55e5789f621f2ce9182f01d9636/flask_restful/__init__.py#L283) swallows custom exception messages right now. You could use `abort`, but theres issues with that too (see #205).